src/main.cpp

```diff
@@ -900,6 +903,20 @@ bool ContextualCheckTransaction(
         }
     }
 
+    // From Canopy onward, coinbase transaction must include outputs corresponding to the
+    // ZIP 207 consensus funding streams active at the current block height. To avoid
+    // double-decrypting, we detect any shielded funding streams during the Heartwood
+    // consensus check. If Canopy is not yet active, fundingStreamElements will be empty.
+    std::set<Consensus::FundingStreamElement> fundingStreamElements;
+    if (canopyActive) {
+        fundingStreamElements = Consensus::GetActiveFundingStreamElements(
+            nHeight,
+            // GetBlockSubsidy must be guarded because it will throw
+            // if called at slow-start period heights.
+            GetBlockSubsidy(nHeight, chainparams.GetConsensus()),
```

Comment:
I'm confused by this comment, because `consensusParams.SubsidySlowStartShift()` is defined as `consensusParams.nSubsidySlowStartInterval / 2`, so the assertion shouldn't ever fire. I guess that the unit tests were configuring regtest mode in a strange way? In any case, guarding this scope on Canopy activation is sensible.

Reply:
The throw happens when `nHeight < consensusParams.SubsidySlowStartShift()` and in some tests `nHeight == 0`. This was the less-disruptive fix; the "proper" fix would probably be to make `GetBlockSubsidy` return a `boost::optional`

Reply:
Hm. I know that I added this guard due to hitting that assertion in GetBlockSubsidy, but perhaps that's in regtest only (when nSubsidySlowStartInterval == 0)